src/Parser/DeathKnight/Blood/Modules/Items/SoulflayersCorruption.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS/index';
+import ITEMS from 'common/ITEMS';
+import ItemDamageDone from 'Main/ItemDamageDone';
+import ItemHealingDone from 'Main/ItemHealingDone';
+import Wrapper from 'common/Wrapper';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+import calculateEffectiveDamage from 'Parser/Core/calculateEffectiveDamage';
+import calculateEffectiveHealing from 'Parser/Core/calculateEffectiveHealing';
+
+class SoulflayersCorruption extends Analyzer {
+
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasChest(ITEMS.SOULFLAYERS_CORRUPTION.id);
+  }
+
+  heal=0;
+  damage=0;
+  absorb=0;
+  umbilicusEternusCount=0;
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.BLOOD_PLAGUE.id) {
+      console.log("heal", this.heal);
```

Comment:
Remember to remove this.

Reply:
ah yes part of my testing lol will remove.